packages/core/src/pubsub/__tests__/outstanding-queries.test.ts

```diff
@@ -0,0 +1,143 @@
+    outstandingQueries.queryQueue.enqueue(q1)
+    outstandingQueries.queryQueue.enqueue(q2)
+    outstandingQueries.queryQueue.enqueue(q3)
+    const size = outstandingQueries.queryQueue.size()
+    expect(size).toEqual(3)
+  })
+
+  test('Prioritized Queue verify queue empty after clean up', async () => {
+    const t1 = new Date('2022-05-12T10:20:30Z').getTime()
+    const q1 = new Query(t1, FAKE_STREAM_ID, 'a')
+    outstandingQueries.queryQueue.enqueue(q1)
+    outstandingQueries.queryMap.set('a', q1)
+    assert(outstandingQueries.queryQueue.front() == q1)
+    assert(outstandingQueries.queryMap.get('a') == q1)
+    console.log('outstandingQueries.queryQueue - before')
+    console.log(outstandingQueries.queryQueue)
+    console.log(outstandingQueries.queryMap)
+    outstandingQueries.cleanUpExpiredQueries()
+    console.log('outstandingQueries.queryQueue - after')
+    console.log(outstandingQueries.queryQueue)
+    console.log(outstandingQueries.queryMap)
+    expect(outstandingQueries.queryQueue.isEmpty()).toEqual(true)
+  })
+
+  test('Prioritized Queue verify map empty after clean up', async () => {
+    const t1 = new Date('2022-05-12T10:20:30Z').getTime()
+    const q1 = new Query(t1, FAKE_STREAM_ID, 'a')
+    outstandingQueries.queryQueue.enqueue(q1)
+    outstandingQueries.queryMap.set('a', q1)
+    assert(outstandingQueries.queryQueue.front() == q1)
+    assert(outstandingQueries.queryMap.get('a') == q1)
+    console.log('outstandingQueries.queryMap - before')
+    console.log(outstandingQueries.queryQueue)
+    console.log(outstandingQueries.queryMap)
+    outstandingQueries.cleanUpExpiredQueries()
+    console.log('outstandingQueries.queryMap - after')
+    console.log(outstandingQueries.queryQueue)
+    console.log(outstandingQueries.queryMap)
+    expect(outstandingQueries.queryMap.size).toEqual(0)
+  })
+})
+
+describe('Outstanding Queries', () => {
+  beforeEach(() => {
+    outstandingQueries.queryMap.clear()
+    outstandingQueries.queryQueue.clear()
+  })
+
+  test('Oustanding Queries, Add Query', async () => {
+    const t1 = new Date('2022-05-12T10:20:30Z').getTime()
+    const q1 = new Query(t1, FAKE_STREAM_ID, 'a')
+    const testQueryID = 'testID'
+    outstandingQueries.add(testQueryID, q1)
+    expect(outstandingQueries.queryMap.size).toEqual(1)
+    expect(outstandingQueries.queryQueue.size()).toEqual(1)
+  })
+
+  test('Oustanding Queries, Remove Query', async () => {
+    const t1 = new Date('2022-05-12T10:20:30Z').getTime()
+    const testQueryID = 'testID'
+    const q1 = new Query(t1, FAKE_STREAM_ID, testQueryID)
+    outstandingQueries.add(testQueryID, q1)
+    assert(outstandingQueries.queryMap.size === 1)
+    assert(outstandingQueries.queryQueue.size() === 1)
+    outstandingQueries.remove(q1)
+    expect(outstandingQueries.queryMap.size).toEqual(0)
+
+    expect(outstandingQueries.queryQueue.size()).toEqual(0)
+  })
+
+  test('Oustanding Queries, cleanUpExpiredQueries', async () => {
```

Comment:
would be good to have a test where some of the queries are expired and others aren't, to make sure it only cleans up the ones that are actually expired.

Reply:
sorry where? I'm not seeing it